### DISCUSSION TITLE
Error when connecting account

### DISCUSSION THREAD

**Discussion by neilfrazerlx** [BottlecapDave/HomeAssistant-OctopusEnergy#190] [Q&A]
Hi,

I'm getting the following error when trying to connect to my octopus account. It worked previously. I restored from a backup after messing up a few things and now I can't connect.

Any thoughts?

Logger: homeassistant.config_entries
Source: custom_components/octopus_energy/api_client.py:193 
Integration: Octopus Energy (documentation, issues) 
First occurred: 10:17:03 (1 occurrences) 
Last logged: 10:17:03

Error setting up entry Account for octopus_energy
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 383, in async_setup
    result = await component.async_setup_entry(hass, self)
  File "/config/custom_components/octopus_energy/__init__.py", line 48, in async_setup_entry
    await async_setup_dependencies(hass, config)
  File "/config/custom_components/octopus_energy/__init__.py", line 112, in async_setup_dependencies
    account_info = await client.async_get_account(config[CONFIG_MAIN_ACCOUNT_ID])
  File "/config/custom_components/octopus_energy/api_client.py", line 183, in async_get_account
    "gas_meter_points": list(map(lambda mp: {
  File "/config/custom_components/octopus_energy/api_client.py", line 190, in <lambda>
    "agreements": list(map(lambda a: {
  File "/config/custom_components/octopus_energy/api_client.py", line 193, in <lambda>
    "tariff_code": a["tariff"]["tariffCode"] if "tariff" in a and "tariffCode" in a["tariff"] else None,
TypeError: argument of type 'NoneType' is not iterable

**Comment by BottlecapDave** [comment]
Hello, sorry that you're receiving this error. This suggests that your account is potentially missing some gas meters. Have you ever had gas meters or have you recently had them removed?. I'll try and update the integration to handle this better.

**Comment by neilfrazerlx** [comment]
Hi David,Thanks for responding. We have had both gas and electric working fine. We are in the process of adding export so it might be that.Neil.On 28 Mar 2023, at 20:00, David Kendall ***@***.***> wrote:﻿
Hello, sorry that you're receiving this error. This suggests that your account is potentially missing some gas meters. Have you ever had gas meters or have you recently had them removed?. I'll try and update the integration to handle this better.

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you authored the thread.Message ID: ***@***.***>
[
{
***@***.***": "http://schema.org",
***@***.***": "EmailMessage",
"potentialAction": {
***@***.***": "ViewAction",
"target": "https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/discussions/190#discussioncomment-5457746",
"url": "https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/discussions/190#discussioncomment-5457746",
"name": "View Discussion"
},
"description": "View this Discussion on GitHub",
"publisher": {
***@***.***": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Reply by BottlecapDave** [reply]
Hello. There's been a few releases since this was raised that's improved account creations. Has this issue been resolved?

**Comment by neilfrazerlx** [comment]
Yes. We’re up and running again. Thanks.Sent from the BatphoneOn 9 Apr 2023, at 09:54, David Kendall ***@***.***> wrote:﻿
Hello. There's been a few releases since this was raised that's improved account creations. Has this issue been resolved?

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you authored the thread.Message ID: ***@***.***>
[
{
***@***.***": "http://schema.org",
***@***.***": "EmailMessage",
"potentialAction": {
***@***.***": "ViewAction",
"target": "https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/discussions/190#discussioncomment-5563220",
"url": "https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/discussions/190#discussioncomment-5563220",
"name": "View Discussion"
},
"description": "View this Discussion on GitHub",
"publisher": {
***@***.***": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]
